Packs/Rapid7AppSec/Integrations/Rapid7AppSec/Rapid7AppSec.py

```diff
@@ -0,0 +1,1270 @@
+                                readable_parser: Callable | None = None,
+                                edit_outputs_settings: AddToOutput | None = None,
+                                ) -> CommandResults:
+    """
+    Create a CommandResult for list commands.
+
+    Args:
+        data (list): Outputs.
+        obj_type (str): The object type.
+        response (dict[str, Any] | list[dict[str, Any]]): Raw response.
+        title (str | None, optional): Title for the readable outputs. Defaults to None.
+        headers (list | None, optional): Headers for the readable output. Defaults to None.
+        readable_parser (Callable | None, optional): Parser command for readable output only. Defaults to None.
+        edit_outputs_settings (ObjectEditor | None, optional): Edit output settings. Defaults to None.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    outputs_key_field = DEFAULT_OUTPUT_KEY_FIELD
+
+    if edit_outputs_settings:
+        if isinstance(edit_outputs_settings, PrefixToResponse):
+            outputs = [edit_outputs_settings.data_to_add | {edit_outputs_settings.prefix: data}]
+            outputs_key_field = edit_outputs_settings.id_key
+        else:
+            outputs = [edit_outputs_settings.data_to_add | obj for obj in data]
+
+    else:
+        outputs = data
+
+    readable_output = create_list_readable_output(
+        data=copy.deepcopy(outputs) if not isinstance(edit_outputs_settings, PrefixToResponse) else copy.deepcopy(data),
+        obj_type=obj_type,
+        title=title,
+        headers=headers,
+        readable_parser=readable_parser
+    )
+    command_results = CommandResults(
+        readable_output=readable_output,
+        outputs_prefix=generate_output_prefix(obj_type),
+        outputs_key_field=outputs_key_field,
+        outputs=outputs,
+        raw_response=response,
+    )
+    return command_results
+
+
+@logger
+def handle_list_request(request_command: Callable,
+                        pagination: Pagination | None = None,
+                        obj_id: str | None = None,) -> dict[str, Any] | list[dict[str, Any]]:
+    """
+    Handle list requests.
+
+    Args:
+        pagination (Pagination | None, optional): Pagination arguments. Defaults to None.
+        request_command (Callable | None, optional): List request command. Defaults to None.
+        obj_id (str | None, optional): The ID of the object to get. Defaults to None.
+
+    Returns:
+        dict[str, Any] | list[dict[str, Any]]: API response from AppSec API.
+    """
+    if obj_id:
+        request_command = partial(request_command,
+                                  obj_id=obj_id)
+    elif pagination:
+        request_command = partial(request_command,
+                                  index=pagination.page and pagination.page - 1,
+                                  size=pagination.page_size or pagination.limit)
```

Comment:
* We should support 2 options for pagination: page & page_size, limit. In the case that all three parameters are given, the limit parameter should be ignored, as you did in the code. But, in your code, it is also possible to get page and limit, and the command will work. This shouldn't be allowed. so, please add a validation to ensure that if a page or page_size is given then the other one is provided as well. You can find more information about the pagination [here](https://xsoar.pan.dev/docs/integrations/code-conventions#pagination-in-integration-commands).
* Is there a limit on the API side for how many results are returned in one request or not? If there is then you should implement a loop of requests until you reach the wanted number of results, as written in the link above in the section of Automatic Pagination.

Reply:
Fixed in the Pagination class :)